src/planner/expand_hypertable.c

```diff
@@ -154,6 +155,27 @@ is_timestamptz_op_interval(Expr *expr)
 		   (c1->consttype == INTERVALOID && c2->consttype == TIMESTAMPTZOID);
 }
 
+static Datum
+int_get_datum(int64 value, Oid type)
+{
+	switch (type)
+	{
+		case INT2OID:
+			return Int16GetDatum(value);
+		case INT4OID:
+			return Int32GetDatum(value);
+		case INT8OID:
+			return Int64GetDatum(value);
+		case TIMESTAMPOID:
+			return TimestampGetDatum(value);
+		case TIMESTAMPTZOID:
+			return TimestampTzGetDatum(value);
+	}
+
+	elog(ERROR, "unsupported datatype in int_get_datum: %s", format_type_be(type));
```

Comment:
Still not sure why we have this error since it is impossible to reach here given the type check already outside this function. Also codecov is complaining.

Reply:
This is a bit of a grey area. Code gets refactored all the time and conditions that are true now might not be later. I prefer to have an error here over just pg_unreachable because an error will only affect the session hitting this while pg_unreachable will kill all sessions and have postgres go into recovery.